pkg/morph/client/netmap/netmap.go

```diff
@@ -107,6 +155,92 @@ func (c *Client) EpochSnapshot(args EpochSnapshotArgs) (*EpochSnapshotValues, er
 	}, nil
 }
 
+func (c *Client) Candidates(_ GetNetMapCandidatesArgs) (*GetNetMapCandidatesValues, error) {
+	prms, err := c.client.TestInvoke(
+		c.netMapCandidatesMethod,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("could not perform test invocation (%s): %w", c.netMapCandidatesMethod, err)
+	}
+
+	candVals, err := peersWithStateFromStackItems(prms, c.netMapCandidatesMethod)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse contract response: %w", err)
+	}
+
+	return candVals, nil
+}
+
+func peersWithStateFromStackItems(stack []stackitem.Item, method string) (*GetNetMapCandidatesValues, error) {
+	if ln := len(stack); ln != 1 {
+		return nil, fmt.Errorf("unexpected stack item count (%s): %d", method, ln)
+	}
+
+	netmapNodes, err := client.ArrayFromStackItem(stack[0])
+	if err != nil {
+		return nil, fmt.Errorf("could not get stack item array from stack item (%s): %w", method, err)
+	}
+
+	res := &GetNetMapCandidatesValues{
+		netmapNodes: make([]*PeerWithState, 0, len(netmapNodes)),
+	}
+
+	for i := range netmapNodes {
+		node, err := peerWithStateFromStackItem(netmapNodes[i])
+		if err != nil {
+			return nil, fmt.Errorf("could not parse stack item (Peer #%d): %w", i, err)
+		}
+
+		res.netmapNodes = append(res.netmapNodes, node)
+	}
+
+	return res, nil
+}
+
+func peerWithStateFromStackItem(prm stackitem.Item) (*PeerWithState, error) {
+	prms, err := client.ArrayFromStackItem(prm)
+	if err != nil {
+		return nil, fmt.Errorf("could not get stack item array (PeerWithState): %w", err)
+	} else if ln := len(prms); ln != peerWithStateFixedPrmNumber {
+		return nil, fmt.Errorf(
+			"unexpected stack item count (PeerWithState): expected %d, has %d",
+			peerWithStateFixedPrmNumber,
+			ln,
+		)
+	}
+
+	var res PeerWithState
+
+	// peer
+	if res.peer, err = peerInfoFromStackItem(prms[0]); err != nil {
+		return nil, fmt.Errorf("could not get bytes from 'node' field of PeerWithState: %w", err)
+	}
+
+	// state
+	if prms[1].Type() != stackitem.IntegerT {
```

Comment:
I think you already discussed that this check is redundant. Big integers can be returned as `ByteArrayT` from the RPC. This is not the case for client status, but in general this can happen with integers and `IntFromStackItem` already covers this case.

Reply:
ok, deleted that check too